geonetwork/core-geonetwork
# Improve workflow process facet in editor board

Currently the facet doesn't track the status properly if the metadata is approved and has a draft copy, displaying always the status **Approved**.

- If the metadata is approved without a working copy or
- the metadata approved with a draft copy or
- the metadata os approved with a draft copy that is submitted for review.

This pull request adds a new facet that takes in account the status of the approved and draft versions of a metadata to display more detailed statuses:

- `Approved`: metadata approved without a working copy
- `Approved with draft copy`: metadata approved with a working copy
- `Approved with draft copy submitted`: metadata approved with a working copy that has been submitted for review
 
![metadata-statuses-with-draft-statuses](https://user-images.githubusercontent.com/1695003/136961441-b495e244-bd1d-4d91-bdf2-0b87d1d9ec1d.png)

---

To test the different values:

1) Login as administrator and enable the workflow in the settings.
2) Create a metadata
3) Create another metadata and submit it for review
4) Create another metadata, submit it for review and approve it
5) Create another metadata, submit it for review, approve it and edit it
6) Create another metadata, submit it for review, approve it, edit it and submit for review
7) In the editor board should be displayed the **Workflow process** similar to the previous screenshot.

---

The pull request is done for `3.12.x`. as seems the facet seem is not available in `main` branch. A feature to port to `main` branch in a separate pull request.
